subsys/bluetooth/audio/codec.c

```diff
@@ -79,8 +158,99 @@ uint8_t bt_audio_codec_cfg_get_val(const struct bt_audio_codec_cfg *codec_cfg, u
 	return param.data_len;
 }
 
+int bt_audio_codec_cfg_set_val(struct bt_audio_codec_cfg *codec_cfg, uint8_t type,
+			       const uint8_t *data, size_t data_len)
+{
+	CHECKIF(codec_cfg == NULL) {
+		LOG_DBG("codec_cfg is NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(data == NULL) {
+		LOG_DBG("data is NULL");
+		return -EINVAL;
+	}
+
+	CHECKIF(data_len == 0U) {
+		LOG_DBG("data_len is 0");
+		return -EINVAL;
+	}
+
+	for (uint16_t i = 0U; i < codec_cfg->data_len;) {
+		const uint8_t len = codec_cfg->data[i++];
+		const uint8_t data_type = codec_cfg->data[i++];
+		const uint8_t value_len = len - sizeof(data_type);
```

Comment:
```suggestion
		const size_t len = codec_cfg->data[i++];
		const uint8_t data_type = codec_cfg->data[i++];
		const size_t value_len = len - sizeof(data_type);
```

Reply:
No reason to set these as `size_t` as the BT spec have declared them to be 8-bit.

Reply:
`data_len` is `size_t` and is compared with `uint8_t` value_len, change `data_len` to `uint8_t` ?

Reply:
Ah, good point. Added a check for this. Usually we use `size_t` in APIs for sizes. Fixed by adding a `data_len > UINT8_MAX` check.